### PR TITLE
feat(table-handle): add `table-handle`

### PR DESCRIPTION
**Pull request by shlroland**
not yet implemented:
1. entire table handle (I feel that this feature should be integrated into the block handle, similar to the notion style.)
2. docs

issues that exist
1. The block handle is not suitable for tables.
2. It appears that the flip function of `floatingui` conflicts with `EditorView#posAtCoords`, and there is an issue with recognizing the handle in the last column.

Additional suggestions
~~1. Should prosemirror-tables be included in the export of @prosekit/pm?~~

@ocavue

### CODE REVIEW

**Comment by ocavue** [comment]
> Should prosemirror-tables be included in the export of @prosekit/pm?

I prefer only putting the most common used packages in `@prosekit/pm`.

**Comment by ocavue** [comment]
This looks nice. Could you also fix the CI issues? You can run `pnpm run build`, `pnpm run lint` and `pnpm run typecheck` locally.

File: packages/core/src/utils/find-parent-node-of-type.ts [hunk 1]
```diff
@@ -12,13 +12,12 @@ export function findParentNodeOfType(
   /**
    * The type of the node to remove.
    */
-  type: NodeType | string,
+  types: NodeType | string | Array<NodeType | string>,
```
Comment (ocavue) [inline packages/core/src/utils/find-parent-node-of-type.ts]: 1. Let's not introduce API breaking changes unless we have to.
2. Is this change related to the `table-handle` feature? I did a quick search and couldn't find its usage in this PR.  If this change is not required by the `table-handle` feature, it's better to open it in another PR.

Comment (shlroland) [reply packages/core/src/utils/find-parent-node-of-type.ts]: I'm sorry, but this is the function I used before, and it was deleted in the final state. I'm ready to restore it to its original state.

**Comment by shlroland** [comment]
> This looks nice. I'll review it soon.
> 
> Meanwhile, could you also fix the CI issues? You can run `pnpm run fix`, `pnpm run build`, and `pnpm run typecheck` locally to verify your code.

There is an interesting observation: when running pnpm run fix on md files in GitHub actions, no errors occur. However, when I run the same command locally, errors occur. What could be the cause of this discrepancy?

**Comment by ocavue** [comment]
I'm not sure. Maybe you can try to remove the whole project, clone it and then do a fresh install.

**Comment by shlroland** [comment]
@ocavue  I noticed that this PR #756  has been merged and made some adjustments. And should we proceed with further development of this PR?

**Comment by ocavue** [comment]
Yes I added some ProseMirror commands in my previous PR, so that we can focus on UI components in this PR.

**Comment by shlroland** [comment]
> Yes I added some ProseMirror commands in my previous PR, so that we can focus on UI components in this PR.

I have already performed a rebase on the main branch.

File: playground/examples/react/table/table-row-handle.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          command={(table) => editor.commands.selectTable({ table })}
+        >
+          <div>select entire table</div>
+        </TableHandlePopoverItem>
+        <TableHandlePopoverItem
+          className={Themes.TABLE_CELL_MENU_ITEM}
+          command={editor.commands.deleteTable}
+        >
+          <div>remove entire table</div>
+        </TableHandlePopoverItem>
```
Comment (ocavue) [inline playground/examples/react/table/table-row-handle.tsx:62]: Let's remove these two menu items. It doesn't feel right to put actions about the whole tables in the row handle.

**Comment by ocavue** [comment]
@shlroland Thanks so much for your contribution. There are some trivial things I'd like to tweak.  I will merge your PR first and do some updates.
